share/picongpu/tests/Synchrotron_plugin/lib/validate.py

```diff
@@ -38,57 +31,84 @@ def read_setup(folderName):
 iterNo = 4000  # iteration from which we read the data must be the same as in ../bin/validate.ci
 
 
-def main(dataPath):
-    print(f"Validating data in {dataPath}")
-    gamma, Heff, dt = read_setup(dataPath)
-
-    series = io.Series(dataPath + "/openPMD/simData_test_%T.bp", io.Access.read_only)
+def read_photon_data(series):
     it = series.iterations[iterNo]
-
-    simulation_dt = it.get_attribute("dt") * it.get_attribute("unit_time")
-    if abs(simulation_dt - dt) > 1e-10:  # check if the simulation dt is the same as the expected dt. 1e-10 is arbitrary
-        raise SystemExit(f"Simulation dt is {simulation_dt} but expected dt is {dt}\n Test failed")
-
     h = it.particles["y"]  # photons
-    e = it.particles["e"]  # electrons
 
-    # photons
+    # Reading photon weights
     h1 = h["weighting"][io.Mesh_Record_Component.SCALAR]
     w = h1[:]
     w_SI = h1.unit_SI
     series.flush()
     w *= w_SI
+
+    # Reading photon momentum
     h1 = h["momentum"]["y"]
     p_y = h1[:]
     p_y_SI = h1.unit_SI
     series.flush()
     p_y = np.float64(p_y)
     p_y *= p_y_SI / w
-    hist_data = np.abs(p_y)
+
+    # Calculate photon energy
+    hist_data = np.abs(p_y * const.c / const.elementary_charge)
+
     if len(hist_data) < 5e5:
         raise SystemExit(
```

Comment:
Just noticed: Do we really want to raise `SystemExit` here? Raising something a little milder like `ValueError` or a custom `Exception` would have the same effect for running this script. But in case anyone ever tries to build something on top of this, a `ValueError` could be caught and handled appropriately.

Reply:
I changed it to ValueError